scalyr_agent/sharded_copying_manager/worker.py

```diff
@@ -501,7 +490,6 @@ def run(self):
                                 >= pipeline_byte_threshold
                                 and self.__pending_add_events_task.next_pipelined_task
                                 is None
-                                and not disable_pipelining
```

Comment:
I am not sure how to be withpipelining. In original  copying manager the function `self.__has_pending_log_changes()` checks for padding new log matches. I guess here we can do the same for new pending log processors and disable pipeline if there are pending logprocessors.

Reply:
I'll have to think through this a little more.  Add a TODO here to say that we need to handle the `disable_pipelining` case.  That was added to solve a very specific bug with respect to how we invoked the callback handlers on the processors.  It may be that with the new design that bug goes away.  Let's save it as one of the last things we need to address for this PR.

Reply:
Let's add this TODO and then follow up in a separate PR to fix any issues.  This is a little bit of a complicated issue and I want to look at the code a little more.